snippets/atom-cloudslang-extension.cson

```diff
@@ -0,0 +1,185 @@
+#   All rights reserved. This program and the accompanying materials
+#   are made available under the terms of the Apache License v2.0 which accompany this distribution.
+#
+#   The Apache License is available at
+#   http://www.apache.org/licenses/LICENSE-2.0
+########################################################################################################
+
+'.source.sl':
+
+  'Flow':
+    'prefix': 'flow'
+    'body': """
+      ####################################################
+      # ${17:Flow description}
+      #
+      # Inputs:
+      #   - ${5:input name} - ${18:input description}
+      # Outputs:
+      #   - ${14:output name} - ${19:output description}
+      # Results:
+      #   - ${16:result name} - ${20:result description}
+      ####################################################
+
+      namespace: ${1:current.namespace}
+
+      imports:
+        ${2:import name}: ${3:import.namespace}
+
+      flow:
+        name: ${4:flow name}
+
+        inputs:
+          - ${5:input name}: ${6:input value}
+
+        workflow:
+          - ${7:task name}:
+              do:
+                ${8:executable name}:
+                  - ${9:executable argument}
+              publish:
+                - ${10:publish name}: ${11:publish value}
+              navigate:
+                ${12:executable result name}: ${13:navigate to}
+
+        outputs:
+          - ${14:output name}: ${15:output value}
+
+        results:
+          - ${16:result name}
+    """
+
+
+  'Operation':
+    'prefix': 'operation'
+    'body': """
+      ####################################################
+      # ${12:Operation description}
+      #
+      # Inputs:
+      #   - ${5:input name} - ${13:input description}
+      # Outputs:
+      #   - ${8:output name} - ${14:output description}
+      # Results:
+      #   - ${10:result name} - ${15:result description}
+      ####################################################
+
+      namespace: ${1:current.namespace}
+
+      operation:
+        name: ${4:operation name}
+
+        inputs:
+          - ${5:input name}: ${6:input value}
+
+        action:
+          python_script: ${7:python script}
+
+        outputs:
+          - ${8:output name}: ${9:output value}
+
+        results:
+          - ${10:result name}: ${11:result condition}
+    """
+
+
+  'simple input':
+    'prefix': 'input'
+    'body': """
+          - ${1:input name}: ${2:input value}
+    """
+
+  'input with properties':
+    'prefix': 'input with properties'
+    'body': """
+          - ${1:input name}:
+              required: ${2:true|false}
+              default: ${3:default value}
+              overridable: ${4:true|false}
+              system_properties: ${5:system.property.key}
+    """
+
+  'output':
+    'prefix': 'output'
+    'body': """
+          - ${1:output name}: ${2:output value}
+    """
+
+  'publish':
+    'prefix': 'publish'
+    'body': """
+          - ${1:publish name}: ${2:publish value}
```

Comment:
oops, wrong line for my comment, I meant here that `${2:import namespace}` (in import snippet) should be `${2:import.namespace}` (like in flow snippet) since value has the form `docker.images.blabla`

on other places I would use underscore (`_`) as separator to be easier to select the values with double click (quick replace)